src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -113,6 +104,57 @@ public AbstractNode createNode(UUID studyUuid, UUID nodeId, AbstractNode nodeInf
         }).orElseThrow(() -> new StudyException(ELEMENT_NOT_FOUND));
     }
 
+    @Transactional
+    public Pair<UUID, UUID> duplicateStudyNode(UUID studyUuid, UUID nodeToCopyUuid, UUID referenceNodeUuid, InsertMode insertMode) {
+        Optional<NodeEntity> referenceNodeOpt = nodesRepository.findById(referenceNodeUuid);
+        NodeEntity referenceNodeEntity = referenceNodeOpt.orElseThrow(() -> new StudyException(NODE_NOT_FOUND));
+
+        Optional<NodeEntity> nodeToCopyOpt = nodesRepository.findById(nodeToCopyUuid);
+        NodeEntity nodeToCopyEntity = nodeToCopyOpt.orElseThrow(() -> new StudyException(NODE_NOT_FOUND));
+
+        UUID newGroupUuid = UUID.randomUUID();
+        UUID modificationGroupUuid = getModificationGroupUuid(nodeToCopyUuid);
+        UUID reportUuid = getReportUuid(nodeToCopyUuid);
+        UUID newReportUuid = UUID.randomUUID();
+        //First we create the modification group
+        networkModificationService.createModifications(modificationGroupUuid, newGroupUuid, reportUuid);
+
+        if (insertMode.equals(InsertMode.BEFORE) && referenceNodeEntity.getType().equals(NodeType.ROOT)) {
+            throw new StudyException(NOT_ALLOWED);
+        }
+        NodeEntity parent = insertMode.equals(InsertMode.BEFORE) || insertMode.equals(InsertMode.NEW_BRANCH) ?
```

Comment:
why '|| insertMode.equals(InsertMode.NEW_BRANCH)'  ?
if we paste a node in a new branch, I think it's in a new branch starting from the referenceNode, like when we create a new node ...